### PR TITLE
NXP-29652: Use an Elasticsearch cluster for Elasticsearch unit tests

### PR DESCRIPTION
**Pull request by ataillefer**

### CODE REVIEW

**Review by bdelbosc — DISMISSED** [review]

**Review by troger — DISMISSED** [review]

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestMapping.java [hunk 1]
```diff
@@ -128,8 +128,13 @@ public void testIlikeSearch() throws Exception {
...
         Assert.assertEquals(0, ret.totalSize());
-        ret = ess.query(new NxQueryBuilder(session).nxql("SELECT * FROM Document WHERE dc:description LIKE 'UPPER%'"));
-        Assert.assertEquals(0, ret.totalSize());
+        // match_phrase_prefix query against a keyword field behaves differently depending on the Elasticsearch version
+        // prior to 6.7.0, it doesn't match
+        // since 6.7.0, it matches, yet we're not sure if it is really expected
+        // since 7, it raises an exception
+        // ret = ess.query(new NxQueryBuilder(session).nxql("SELECT * FROM Document WHERE dc:description LIKE
+        // 'UPPER%'"));
+        // Assert.assertEquals(0, ret.totalSize());
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestMapping.java:137]: What's the point of keeping commented code/test here? It will be never executed/run.

Comment (ataillefer) [reply modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestMapping.java:137]: It's to illustrate this interesting mapping edge case and make it easily testable against an ES cluster of the right version.
@bdelbosc WDYT?

Comment (bdelbosc) [reply modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestMapping.java:137]: we don't mind, this is a temporary code until the es7 upgrade is merged

**Review by bdelbosc — APPROVED** [review]

**Review by troger — APPROVED** [review]
